### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by AVyha**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+        if (args, power) in data:
+            print("Getting from cache")
+            return data[args, power]
+        else:
+            print("Calculating new result")
+            if power == 0:
+                data[args, power] = func(*args)
+            else:
+                data[args, power] = func(*args, power)
+            return data[args, power]
```
Comment (linaSam) [inline app/main.py:14]: you have the same return in line 7 . change your code so that it will have return only once

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,15 @@
 def cache(func):
-    # Write your code here
-    pass
+    data = {}
+
+    def inner(*args, power: int = 0):
```
Comment (linaSam) [inline app/main.py:4]: variable "power" doesn't really do anything. try rewrite your code without it.

**Review by linaSam — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (AVyha) [reply app/main.py:4]: thank you, i wrote this variable for test myself this decorator and forget to clean

**Review by linaSam — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (linaSam) [reply app/main.py]: it's classic one) you are welcome!

File: app/main.py [hunk 3]
```diff
@@ -1,15 +1,12 @@
 def cache(func):
     data = {}
 
-    def inner(*args, power: int = 0):
-        if (args, power) in data:
+    def inner(*args):
+        if args in data:
             print("Getting from cache")
-            return data[args, power]
+            return data[args]
```
[the remark this answers is not in the document]
Comment (AVyha) [reply app/main.py]: i hope in this time all be good) thak you for the review

File: app/main.py [hunk 4]
```diff
@@ -4,9 +4,8 @@ def cache(func):
     def inner(*args):
         if args in data:
             print("Getting from cache")
-            return data[args]
         else:
             print("Calculating new result")
             data[args] = func(*args)
-            return data[args]
+        return data[args]
```
Comment (linaSam) [inline app/main.py:10]: yep, everything great now) nice work!

**Review by linaSam — APPROVED** [review]

**Review by TPolina — APPROVED** [review]
Well done!

**Pull request closed without merging**